src/context.c

```diff
@@ -485,7 +523,14 @@ ly_ctx_set_options(struct ly_ctx *ctx, uint16_t option)
 
     if (!(ctx->flags & LY_CTX_SET_PRIV_PARSED) && (option & LY_CTX_SET_PRIV_PARSED)) {
         ctx->flags |= LY_CTX_SET_PRIV_PARSED;
-        if ((lyrc = lys_recompile(ctx, 0))) {
+        if (ctx->flags & LY_CTX_EXPLICIT_COMPILE) {
+            /* compile it now so that the priv pointers are set */
+            lyrc = ly_ctx_compile(ctx);
```

Comment:
does this work? Isn't the logic reverse here? Let's say that the context was already compiled, so there is no `to_compile` set, so this call will do nothing but we would like to recompile. If the context wasn't compiled, we probably don't need to compile it now, just by setting the option.

Reply:
You are right, that was wrong. I have kept it the way it was so that only previously compiled modules will be recompiled, which is probably the best thing to do, anyway.